baybe/utils/dataframe.py

```diff
@@ -413,3 +414,44 @@ def fuzzy_row_match(
             inds_matched.extend(inds_found)
 
     return pd.Index(inds_matched)
+
+
+def pretty_print_dataframe(df: pd.DataFrame) -> str:
```

Comment:
As @AVHopp wished: here are some screenshots to help you compare and choose between the two printing versions. @AdrianSosic @Scienfitz 
<img width="430" alt="current_pretty_print" src="https://github.com/emdgroup/baybe/assets/118910837/a10f2cf6-1d63-4fde-9f24-20954705835d">
<img width="435" alt="current_pretty_print2" src="https://github.com/emdgroup/baybe/assets/118910837/c9363cf5-7560-4897-a81c-cd42d09b10fb">
<img width="304" alt="standard_version" src="https://github.com/emdgroup/baybe/assets/118910837/1512bb03-2675-40de-90da-aeb26c13f045">
<img width="727" alt="standard_version2" src="https://github.com/emdgroup/baybe/assets/118910837/b5868548-2b4e-4bf8-9927-a4392fc9676e">

Reply:
yep its clearly better than the previous versions, but ideally we can achieve something like it without additional dependency